gms/endpoint_state.hh

```diff
@@ -19,7 +19,7 @@
 
 namespace gms {
 
-using application_state_map = std::map<application_state, versioned_value>;
+using application_state_map = std::unordered_map<application_state, versioned_value>;
```

Comment:
Why not `absl::flat_hash_map`? It's way more efficient than `std::unordered_map`?

I wonder why it's still not the default choice for dictionary data structure in the Scylla code.

Reply:
Great series, a lot of cleanups, thanks!

Reply:
> Why not `absl::flat_hash_map`? It's way more efficient than `std::unordered_map`?
> 
> I wonder why it's still not the default choice for dictionary data structure in the Scylla code.

I don't know if idl supports it.

> Great series, a lot of cleanups, thanks!

Thanks.